macOS CI build fails: hdiutil create Resource busy during DMG creation

## Problem

The macOS build job in CI fails at the DMG creation step with:

```
hdiutil: create failed - Resource busy
Error: Process completed with exit code 1.
```

The PyInstaller build itself completes successfully, but the subsequent `hdiutil create` call fails because the build output directory is still locked/busy.

## Additional note

PyInstaller also emits a deprecation warning:
```
DEPRECATION: Onefile mode in combination with macOS .app bundles (windowed mode) don't make sense and clashes with macOS's security. Please migrate to onedir mode. This will become an error in v7.0.
```

## Suggested fix

- Add a short `sleep` or `sync` before the DMG creation step to ensure file handles are released
- Alternatively, retry `hdiutil create` with a backoff
- Consider migrating from `--onefile` to `--onedir` mode for macOS as recommended by PyInstaller

## Severity
P2 — Medium, blocks macOS release artifacts

--- a/scripts/build_macos.sh
+++ b/scripts/build_macos.sh
@@ -9,7 +9,7 @@ if [ -f "assets/icon.icns" ]; then
 fi
 
 uv run pyinstaller \
-    --onefile \
+    --onedir \
     --windowed \
     --name Jinkies \
     ${ICON_FLAG} \
@@ -18,8 +18,12 @@ uv run pyinstaller \
 
 echo "==> Creating DMG..."
 if command -v hdiutil &> /dev/null; then
+    # Ensure all file handles from the build are released
+    sync
+    sleep 2
+
     mkdir -p dist/dmg_staging
-    cp -r dist/Jinkies.app dist/dmg_staging/ 2>/dev/null || cp dist/Jinkies dist/dmg_staging/
+    cp -r dist/Jinkies.app dist/dmg_staging/ 2>/dev/null || cp -r dist/Jinkies dist/dmg_staging/
     hdiutil create -volname "Jinkies" -srcfolder dist/dmg_staging -ov -format UDZO dist/Jinkies.dmg
     rm -rf dist/dmg_staging
     echo "==> Done: dist/Jinkies.dmg"

